src/gui/main.cpp

```diff
@@ -261,6 +261,12 @@ int main(int argc, char **argv)
 
         platform->setApplication(&app);
 
+        // Check if the user upgraded or downgraded. We do this as early as possible, to detect
+        // a possible downgrade.
+        if (!OCC::Application::configVersionMigration()) {
```

Comment:
Hmm line 260 already accesses the cfg file.
Can we move the check even before that? I guess we have to parse the options for --confdir. (Off topic: we should probably not execute unrelated code in parseOptions).

I guess we can then move the whole function here, and give it a name that represents its current purpose.
(We will no longer need the qtimer connected to quit)

Reply:
done